Models/global_config.hpp

```diff
@@ -13,63 +13,127 @@ namespace config
+                constexpr DSHOTIID(){}
+            };
+        };
+    class RCInputDriver
+    {
+        public:
+        struct RCInputIID{};
+    };
+        class TempPPMDriver : public RCInputDriver
+        {
+            public:
+            struct PPMIID : public RCInputDriver::RCInputIID{
+                constexpr PPMIID(){};
+            };
+        };
+        class TempSBusDriver : public RCInputDriver
+        {
+            public:
+            struct SBusIID : public RCInputDriver::RCInputIID{
+                constexpr SBusIID(){};
+            };
+        };
+    class UnifiedSensorDriver{};
+        class TempVN300Driver : public UnifiedSensorDriver{};
+        class TempVN310Driver : public UnifiedSensorDriver{};
+    class GPSDriver{};
+        class TempNEOM8Driver : public GPSDriver{};
+        class otherGPSDriver : public GPSDriver{};
+    class IMUDriver{};
+        class TempMPU6050Driver : public IMUDriver{};
+    class AirspeedDriver{};
+        class someAirspeedDriver : public AirspeedDriver{};
+     
     //Define factory function to create motor drivers.
     //The driver class must be copy-constructable as the returned object will be a copy.
-    template <class DriverType>
-    MotorDriver constructDriver() {
-        DriverType driver;
+    template <class BaseDriverType, class DerivedDriverType>
+    BaseDriverType constructDriver() {
+        DerivedDriverType driver;
         return driver;
     }
 
     //Typedef pointer to factory function
-    typedef MotorDriver (*DriverFactory)(void);
+    template <class BaseDriverType>
+    using DriverFactory = BaseDriverType (*)(void);
 
     typedef struct {
         ControlAxis_t axis;
-        bool isInverted;
-        DriverFactory driverConstructor;
-        //TODO: Consider a more specific datatype for interfaceID
-        //(e.g. a datatype defined within the motor driver class)
-        uint8_t interfaceID;
+        bool isInverted = false;
+        DriverFactory<MotorDriver> driverConstructor;
+        MotorDriver::MotorIID interfaceID;
     } Motor_t;
 
     typedef struct {
-        bool isEnabled;
-        float p;
-        float i;
-        float d;
+        //TODO: determine other config fields relevant to the RC input type
+        DriverFactory<RCInputDriver> driverConstructor;
+        RCInputDriver::RCInputIID interfaceID;
+    } RCInput_t;
+
+    typedef struct {
+        bool isEnabled = false;
+        float p = 0.0f;
+        float i = 0.0f;
+        float d = 0.0f;
     } AxisPID_t;
 
     typedef struct {
-        AxisPID_t yawPID;
-        AxisPID_t pitchPID;
-        AxisPID_t rollPID;
-        AxisPID_t thrustPID;
+        AxisPID_t yawPID = {};
+        AxisPID_t pitchPID = {};
+        AxisPID_t rollPID = {};
+        AxisPID_t thrustPID = {};
     } ControlPID_t;
 
     typedef struct {
-        float min;
-        float max;
+        float min = 0.0f;
+        float max = 100.0f;
```

Comment:
I think [-90, 90] might be more reasonable default limits, assuming each value is in degrees

Reply:
I intended for these to be percentages, which could be interpreted as needed depending on the flight mode.

Reply:
I'm not sure how percentages would be interpreted by a flight mode... 100% of 360deg? I'm fine with this if we can communicate that it is a percentage. I wouldn't be opposed to adding a percentage type maybe?